File: app/Models/Reserva.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Reserva extends Model
{
    protected $table = 'events'; // <- aquí se indica la tabla real

    protected $fillable = [
        'consultorio',
        'fecha',
        'hora_inicio',
        'hora_fin',
        'user_id',
    ];

    protected $casts = [
        'fecha' => 'datetime',
        'hora_inicio' => 'datetime',
        'hora_fin' => 'datetime',
        'start' => 'datetime',
        'end' => 'datetime',
    ];
    public function toArray()
    {
        return [
            'id' => $this->id,
            'consultorios' => optional($this->consultorio)->nombre ?? 'Sin nombre',
            'fecha' => optional($this->start)->format('Y-m-d'),
            'hora_inicio' => optional($this->start)->format('H:i:s'),
            'hora_fin' => optional($this->end)->format('H:i:s'),
        ];
    }
    public function consultorio()
    {
        return $this->belongsTo(Consultorio::class, 'consultorio_id');
    }
}

File: siscitas_flutter/lib/models/reserva.dart
class Reserva {
  final int id;
  final String consultorio;
  final DateTime fecha;
  final DateTime horaInicio;
  final DateTime horaFin;

  Reserva({
    required this.id,
    required this.consultorio,
    required this.fecha,
    required this.horaInicio,
    required this.horaFin,
  });

factory Reserva.fromJson(Map<String, dynamic> json) {
  return Reserva(
    id: json['id'],
    consultorio: json['consultorio'] ?? 'Sin nombree',
   fecha: json['fecha'] != null
    ? DateTime.parse(json['fecha'] + ' 00:00:00')
    : DateTime(2000, 1, 1),
horaInicio: (json['fecha'] != null && json['hora_inicio'] != null)
    ? DateTime.parse('${json['fecha']} ${json['hora_inicio']}')
    : DateTime(2000, 1, 1, 8, 0),
horaFin: (json['fecha'] != null && json['hora_fin'] != null)
    ? DateTime.parse('${json['fecha']} ${json['hora_fin']}')
    : DateTime(2000, 1, 1, 9, 0),
  );
}
}
